docs: Document how to store xCom objects in S3 (#586)
stackabletech/airflow-operator: docs/modules/airflow/pages/usage-guide/using-kubernetes-executors.adoc

Now you should be able to fetch and inspect logs in the Airflow Web UI from S3 for each DAG run.

image::airflow_dag_s3_logs.png[Airflow DAG S3 logs]

== Store xComs objects in object storage (e.g. S3)

By default Airflow stores the xCom (cross-communications) objects in the Airflow database (often times PostgreSQL), which can cause the database to run out of disk space (especially if you have high xCom traffic).
To prevent this, you can configure Airflow to do the xCom exchange via an object store backend (such as S3) and not the database.
You can read on details in https://airflow.apache.org/docs/apache-airflow/stable/core-concepts/xcoms.html#object-storage-xcom-backend[the official documentation].

[TIP]
====
Airflow stores the xCom objects in the Postgres table `xcom` (e.g. `airflow.public.xcom`).
You can have a look at that table to see what exactly is getting stored.

To e.g. determine the biggest xCom objects you can use
[source,sql]
----
select
	length(value) as size,
	dag_run_id,
	task_id,
	map_index,
	key,
	dag_id,
	run_id,
	"timestamp"
from xcom
order by length(value) desc limit 20;
----

*Being careful* you can also clean up the table to save space in the database.

> You need to run something like `VACUUM FULL` for the changes to take effect on the PostgreSQL disk.

Reviewer: Should we add a little certainty?
Or comment it out in the docs until we know?

Author: The described way worked perfectly fine in our prod env :)
I formulated it a bit carefully to not get blamed when a vacuum full takes some table locks and causes any sort of problem (not a postgres expert)

Reviewer: In that case, we could make it a little less flippant:

```suggestion
In our testing, running `VACUUM FULL` was enough for the changes to take effect on the PostgreSQL disk.
```
